### PR TITLE
Prevent Docker Image Push for Dependabot PR's

### PR DESCRIPTION
**Pull request by bastianschaffer**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -165,6 +165,7 @@ jobs:
       - integration-test
       - image-scan
     runs-on: ubuntu-22.04
+    if: ${{ ! startsWith(github.ref, 'dependabot/')  && ! startsWith(github.head_ref, 'dependabot/')}}
```
Comment (alexanderkiel) [inline .github/workflows/build.yml:168]: Do we need both checks here?

**Review by alexanderkiel — CHANGES_REQUESTED** [review]
Please change the title of the PR and Commit in something like "Prevent Docker Image Push for Dependabot PR's". Simply saying that we restrict the image push action somehow is to general. In the text part of the messages, it would be also good to explain the problem we had with the push action.

File: .github/workflows/build.yml (hunk 1 above)
Comment (bastianschaffer) [reply .github/workflows/build.yml:168]: in a push, the github.ref is the fully formated branch name like `refs/heads/<branch_name>` (so i just realized that i have to add the "refs/heads/" part in order for it to work when something is pushed.
In a pull request, the github.ref is the merge branch like `refs/pull/<pr_number>/merge ` so this check doesn't work in pull requets.

head_ref is the source branch name of the pull request, but it only works in pull requests.

See: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

So if we want to have the check in both pushes and pull requests, both checks are needed. But if we only want to check in pull requests, the right part would be enough.

Comment (alexanderkiel) [reply .github/workflows/build.yml:168]: As the dependant is doing only PR's the second check will be sufficient.

**Review by alexanderkiel — APPROVED** [review]

**Pull request merged**